### PR TITLE
Add Go client

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Part of #163

### CODE REVIEW

**Comment by tenzen-y** [comment]
@danielvegamyhre We must add `doc.go` to generate appropriate codes.

https://github.com/kubernetes-sigs/kueue/blob/a3c76a537dfd68bb49cd661c6c3e7e1044c375f2/apis/kueue/v1beta1/doc.go

**Comment by danielvegamyhre** [comment]
@tenzen-y thanks, I thought that was a generated file when I saw it at first. I've added it and re-ran the generate command but the client-go files are still empty. Do you know if the package structure `apis/<groupName>/<version>/*.go` is required? This old [article](https://cloud.redhat.com/blog/kubernetes-deep-dive-code-generation-customresources) linked in the code-generator README uses that convention and so does Kueue, so I'm wondering that that's the issue

File: api/v1alpha1/jobset_types.go [hunk 1]
```diff
@@ -93,6 +93,7 @@ type ReplicatedJobStatus struct {
 // +kubebuilder:printcolumn:name="Restarts",JSONPath=".status.restarts",type=string,description="Number of restarts"
 // +kubebuilder:printcolumn:name="Completed",type="string",priority=0,JSONPath=".status.conditions[?(@.type==\"Completed\")].status"
 // +kubebuilder:printcolumn:name="Age",JSONPath=".metadata.creationTimestamp",type=date,description="Time this JobSet was created"
+// +genclient
```
Comment (tenzen-y) [inline api/v1alpha1/jobset_types.go:96]:
```suggestion
// +genclient

```

We need a new line.

Comment (danielvegamyhre) [reply api/v1alpha1/jobset_types.go]: Yeah just noticed this is mentioned in the article as well - I added an empty line and moved the `// +genclient` to the top above the kubebuilder tags, following the pattern from the Kueue PR but still no luck.

**Comment by tenzen-y** [comment]
> @tenzen-y thanks, I thought that was a generated file when I saw it at first. I've added it and re-ran the generate command but the client-go files are still empty. Do you know if the package structure `apis/<groupName>/<version>/*.go` is required? This old [article](https://cloud.redhat.com/blog/kubernetes-deep-dive-code-generation-customresources) linked in the code-generator README uses that convention and so does Kueue, so I'm wondering that that's the issue

IIUC, we need to follow the dir structure.

https://github.com/kubernetes/code-generator/blob/65c70a51ae8eca63d4b9ab1e80fe636097a6ce63/generate-groups.sh#L31

File: api/v1alpha1/jobset_types.go (hunk 1 above)
Comment (tenzen-y) [reply api/v1alpha1/jobset_types.go]: What a timing! Nice :)

**Comment by danielvegamyhre** [comment]
> > @tenzen-y thanks, I thought that was a generated file when I saw it at first. I've added it and re-ran the generate command but the client-go files are still empty. Do you know if the package structure `apis/<groupName>/<version>/*.go` is required? This old [article](https://cloud.redhat.com/blog/kubernetes-deep-dive-code-generation-customresources) linked in the code-generator README uses that convention and so does Kueue, so I'm wondering that that's the issue
> 
> IIUC, we need to follow the dir structure.
> 
> https://github.com/kubernetes/code-generator/blob/eec869e1b034acd494e70dee51edafac48a055f1/generate-groups.sh#L30-L32

Yep changing the directory structure worked! Thanks

**Comment by ahg-g** [comment]
Thanks @danielvegamyhre; this PR generates the client libraries for go, what is the next step to generate the apis in python?

**Comment by danielvegamyhre** [comment]
@ahg-g I asked in the api machinery slack about generating protos from our CRD (which we could use to generate the Python types) but was told that this is [not supported]https://kubernetes.slack.com/archives/C0EG7JC6T/p1685662364273809?thread_ts=1685647927.474879&cid=C0EG7JC6T), so the only option I can think of would be to write the protos by hand. This is doable but we will need to manually update the protos every time we make a change to the API, which will be more error prone than the the process being automated as CI step. What do you think?

**Pull request merged**